plugins/catalog/src/alpha/contextMenuItems.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  EntityContextMenuItemBlueprint,
+  useEntityPermission,
+} from '@backstage/plugin-catalog-react/alpha';
+import FileCopyTwoToneIcon from '@material-ui/icons/FileCopyTwoTone';
+import BugReportIcon from '@material-ui/icons/BugReport';
+import CancelIcon from '@material-ui/icons/Cancel';
+import useCopyToClipboard from 'react-use/esm/useCopyToClipboard';
+import { alertApiRef, useApi, useRouteRef } from '@backstage/core-plugin-api';
+import {
+  dialogApiRef,
+  useTranslationRef,
+  type DialogApiDialog,
+} from '@backstage/frontend-plugin-api';
+import { catalogTranslationRef } from './translation';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import {
+  UnregisterEntityDialog,
+  useEntity,
+} from '@backstage/plugin-catalog-react';
+import { rootRouteRef, unregisterRedirectRouteRef } from '../routes';
+import { catalogEntityDeletePermission } from '@backstage/plugin-catalog-common/alpha';
+
+export const copyEntityUrlContextMenuItem = EntityContextMenuItemBlueprint.make(
+  {
+    name: 'copy-entity-url',
+    params: {
+      icon: <FileCopyTwoToneIcon fontSize="small" />,
+      useProps: () => {
+        const [copyState, copyToClipboard] = useCopyToClipboard();
+        const alertApi = useApi(alertApiRef);
+        const { t } = useTranslationRef(catalogTranslationRef);
+
+        React.useEffect(() => {
+          if (!copyState.error && copyState.value) {
+            alertApi.post({
+              message: t('entityContextMenu.copiedMessage'),
+              severity: 'info',
+              display: 'transient',
+            });
+          }
+        }, [copyState, alertApi, t]);
+
+        return {
+          title: t('entityContextMenu.copyURLMenuTitle'),
+          onClick: async () => {
+            copyToClipboard(window.location.toString());
+          },
+        };
+      },
```

Comment:
Yep I kinda like this pattern. I wonder if we should allow all props to be defined both at top-level and `useProps` 🤔, we can leave that for later iteration though as I think we need a bit of use before it can reach it's final form